Tasks/Gradle/CodeAnalysis/gradlesonar.ts

```diff
@@ -0,0 +1,59 @@
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../../definitions/sonarqube-common.d.ts" />
+
+import path = require('path');
```

Comment:
@jasholl This PR seems to contain 92 commits and is touching on logic that you did not write. Would be good to send a more focused PR.